src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -289,6 +289,46 @@ static bool matchMetadata(BSONObj& md1, BSONObj& md2)
     STR_FIELD(md1, ENT_ATTRS_MD_NAME) == STR_FIELD(md2, ENT_ATTRS_MD_NAME);
 }
 
+/* ****************************************************************************
+*
+* equalMetadataValues -
+*
+*/
+static bool equalMetadataValues(BSONObj& md1, BSONObj& md2)
+{
+
+  if (md1.getField(ENT_ATTRS_MD_VALUE).type() != md2.getField(ENT_ATTRS_MD_VALUE).type())
+  {
+    return false;
+  }
+
+  switch (md1.getField(ENT_ATTRS_MD_VALUE).type())
+  {
+    /* FIXME not yet
+    case Object:
+      ...
+      break;
+
+    case Array:
+      ...
+      break;
+    */
+
+    case NumberDouble:
+      return md1.getField(ENT_ATTRS_MD_VALUE).Number() == md2.getField(ENT_ATTRS_MD_VALUE).Number();
+
+    case Bool:
+      return md1.getBoolField(ENT_ATTRS_MD_VALUE) == md2.getBoolField(ENT_ATTRS_MD_VALUE);
+
+    case String:
+      return STR_FIELD(md1, ENT_ATTRS_MD_VALUE) == STR_FIELD(md2, ENT_ATTRS_MD_VALUE);
+
+    default:
+      LM_E(("Runtime Error (unknown metadata value type in DB: %d)", md1.getField(ENT_ATTRS_MD_VALUE).type()));
```

Comment:
Yes, but I think it doesn't worth the effort. This error is supossed not to happen and, even in the rare case it happen, you can check the BSON reference to know to which type it refers.

NTC